client/my-sites/earn/refer-a-friend/index.tsx

```diff
@@ -0,0 +1,159 @@
+/**
+ * External dependencies
+ */
+import { connect } from 'react-redux';
+import React, { FunctionComponent, Fragment, useState, useEffect } from 'react';
+import { compact } from 'lodash';
+
+/**
+ * Internal dependencies
+ */
+import wp from 'lib/wp';
+import { SiteSlug } from 'types';
+import { useTranslate } from 'i18n-calypso';
+import { isJetpackSite } from 'state/sites/selectors';
+import { getSelectedSiteSlug } from 'state/ui/selectors';
+import getSiteBySlug from 'state/sites/selectors/get-site-by-slug';
+import isSiteAutomatedTransfer from 'state/selectors/is-site-automated-transfer';
+import PromoSection, { Props as PromoSectionProps } from 'components/promo-section';
+import { bumpStat, composeAnalytics, recordTracksEvent } from 'state/analytics/actions';
+import ClipboardButtonInput from 'components/clipboard-button-input';
+import { CtaButton } from 'components/promo-section/promo-card/cta';
+
+/**
+ * Image dependencies
+ */
+import earnSectionImage from 'assets/images/earn/earn-section.svg';
+import referralImage from 'assets/images/earn/referral.svg';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+interface ConnectedProps {
+	siteId: number;
+	selectedSiteSlug: SiteSlug;
+	isJetpack: boolean;
+	isAtomicSite: boolean;
+	trackCtaButton: ( feature: string ) => void;
+}
+
+const wpcom = wp.undocumented();
+
+const ReferAFriendSection: FunctionComponent< ConnectedProps > = ( {
+	isJetpack,
+	isAtomicSite,
+	trackCtaButton,
+} ) => {
+	const translate = useTranslate();
+	const [ peerReferralLink, setPeerReferralLink ] = useState( '' );
+
+	useEffect( () => {
+		if ( peerReferralLink ) return;
+		wpcom.me().getPeerReferralLink( ( error: string, data: string ) => {
+			setPeerReferralLink( ! error && data ? data : '' );
+		} );
+	}, [ peerReferralLink ] );
+
+	const onPeerReferralCtaClick = () => {
+		if ( peerReferralLink ) return;
+		wpcom.me().setPeerReferralLinkEnable( true, ( error: string, data: string ) => {
+			setPeerReferralLink( ! error && data ? data : '' );
+		} );
+	};
+
+	const getPeerReferralsCard = () => {
+		const isJetpackNotAtomic = isJetpack && ! isAtomicSite;
+
+		if ( isJetpackNotAtomic ) {
+			return;
+		}
+
+		const cta: CtaButton = {
+			text: translate( 'Get Shareable Link' ) as string,
+			action: () => {
+				trackCtaButton( 'peer-referral-wpcom' );
+				onPeerReferralCtaClick();
+			},
+		};
+
+		if ( peerReferralLink ) {
+			cta.component = <ClipboardButtonInput value={ peerReferralLink } />;
+		}
+
+		return {
+			title: translate( 'Refer a friend, you’ll both earn credits!' ),
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **19** times:
`translate( 'Refer a friend, you’ll both earn credits' )` *ES Score: 12*

Reply:
I need the punctuation.